src/auth/controllers/auth.controller.ts

```diff
@@ -0,0 +1,58 @@
+import {
+    Controller,
+    Post,
+    Body,
+    Res,
+    HttpStatus
+} from '@nestjs/common';
+
+import { AuthService } from '../services/auth.service';
+
+import {
+    CreateUserDto,
+    SignInUserDto
+} from '../dto/auth.dto';
+
+import {
+    createUserSchema,
+    signInUserSchema
+} from '../schema/validation.schema';
+
+@Controller('users')
+export class AuthController {
+
+    constructor(private readonly usersService: AuthService) { }
+
+    @Post('signup')
+    async signUp(@Body() createUserDto: CreateUserDto, @Res() res) {
```

Comment:
not bad, but controller usually will look like this 
```
@Post('/login')
  @ApiOperation({ summary: 'Login for users' })
  @ApiOkResponse({
    status: 200,
    type: LoginResponseDto,
  })
  @ApiResponse({ status: 403, description: 'Forbidden.' })
  @ApiNotFoundResponse({ status: 404, description: 'Not Found.' })
  @ApiNotFoundResponse({ status: 403, description: 'Forbidden' })
  login(
    @Request() request,
    @Body(new JoiValidationPipe(userLoginSchema)) userData: LoginDto,
  ): Promise<LoginResponseDto> {
    return this.usersService.login(userData);
  }
```
I want to say that logic that related to validation you should keep in validation middleware  in Nest it usually pipe